test/EventListener/JsonApiProblemExceptionListenerTest.php

```diff
@@ -135,19 +134,15 @@ public function it_parses_a_debuggable_api_problem_response(): void
         $apiProblem->toArray()->willReturn($data);
         $exception = new \RuntimeException();
 
-        $this->event->getException()->willReturn($exception);
         $this->exceptionTransformer->accepts($exception)->willReturn(true);
         $this->exceptionTransformer->transform($exception)->willReturn($apiProblem->reveal());
 
         $this->request->getRequestFormat()->willReturn('json');
         $this->request->getContentType()->willReturn('application/json');
 
-        $this->event->setResponse(Argument::that(function (JsonResponse $response) use ($data) {
-            return 500 === $response->getStatusCode()
-                && 'application/problem+json' === $response->headers->get('Content-Type')
-                && $response->getContent() === json_encode($data);
-        }))->shouldBeCalled();
+        $this->event->setResponse(new JsonResponse($data, 500, ['Content-Type' => 'application/problem+json']));
```

Comment:
The listener should be setting this response, not the unit test. So this might lead to false positives.
The original code was validating the content of the response, not setting it.
Validation on the header and content is missing.